File: backend/src/main/java/com/gxs/bank/service/AuditLogService.java
package com.gxs.bank.service;

import com.gxs.bank.model.AuditLog;
import com.gxs.bank.repository.AuditLogRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class AuditLogService {
    private final AuditLogRepository auditLogRepository;

    public void logAction(UUID performedBy, AuditLog.Action action, String entityType, UUID entityId, String details) {
        AuditLog log = AuditLog.builder()
            .performedBy(performedBy)
            .action(action)
            .targetEntityType(entityType)
            .targetEntityId(entityId)
            .details(details)
            .build();
        auditLogRepository.save(log);
    }

    public List<AuditLog> getAllLogs() {
        return auditLogRepository.findAll();
    }
}

File: backend-python/com/gxs/bank/service/AuditLogService.py
from com.gxs.bank.model.AuditLog import AuditLog
from com.gxs.bank.repository.AuditLogRepository import AuditLogRepository


class AuditLogService:
    def __init__(self, db):
        self.db = db
        self.auditLogRepository = AuditLogRepository(db)

    def logAction(self, performedBy: str, action, entityType: str, entityId: str, details: str) -> None:
        log = AuditLog(
            performedBy=performedBy,
            action=action,
            targetEntityType=entityType,
            targetEntityId=entityId,
            details=details,
        )
        self.auditLogRepository.save(log)
        self.db.commit()

    def getAllLogs(self):
        return self.auditLogRepository.findAll()
